src/collective/taxonomy/indexer.py

```diff
@@ -39,7 +41,7 @@ def __call__(self):
 
         found = []
         stored_element = getattr(self.context, self.field_name)
-        if not isinstance(stored_element, list):
+        if not isinstance(stored_element, collections.Iterable):
```

Comment:
So I guess you had a set in stored_element, so it was doing stored_element=[set(...)]
Good catch!